### ISSUE TITLE
Select Dice

### ISSUE THREAD

**Issue by ManvilleFog** [buttonmen-dev/buttonmen#1388]
Select Dice: 
Created for the Fightball set by Cheapass games: All Fightball buttons have seven or eight Select dice in their recipe.  Before the start of the game, and after losing a match, choose five Select dice to be part of your die pool.  For this site: Select dice will not necessarily be part of every die in a button's recipe, and the number of dice selected will not always be five. 

Select Dice were once part of the Fightball set, but have since been replaced with the Take Five rule that applies to each button as a whole.  Recent discussions on other issues ( #513 , #1351 ) have made it seem that it might be a good idea to add select dice again.  This would use button containers (thanks @jl8e - I understand what you meant by that now) to isolate the dice available for selection.  In the case of Fightball, every die on the button (plus Auxiliary or other added dice) will be added to the container.  _The reason to do this rather than just use the button special for Fightball would be to increase functionality to be used with future fanatics and fan sets._  This will, for example, provide a way to combine Plasma and Option, and in fact could potentially be used in the background to implement Plasma (and could have been used for option and other variable dice).  In use, the difference between Select and Sideboard dice is that Sideboard dice start out-of-play and Select dice need to be chosen before the start of the game. 

An example from Fightball (if it was not using a button special instead):

```
'Domino',            '5[(4) (4) (8) (8) (8) (10) (12)]'
```

An example from a potential 50 States button:

```
'California',      '@(10) @(20) 1[p(20) q(12)] (Y)? (Z)'
```

---

This is my take on Select Dice.  Rules and dice codes are, of course, up for discussion and clarification.  Also, I'll try to track down where I read that Take Five had replaced Select Dice.  It's entirely possible that it's an assumption I made from learning that Fightball originally used Select Dice and now the only rules I can find simply say "Choose Five" (http://cheapass.com/node/39). 

The BIG question is: should we reintroduce these?

**Comment by TheOrgg** [comment]
Question on formatting of Select based characters.

Should Domino from Fightball be written as "[5] S(4) S(4) S(8) S(8) S(8) S(10) S(12)", Rocky Road (Ice Cream) be (5) (10) (13) [2] St(8) SD(11) Sz(16) SB(20) and Orange Sherbert be (4) (6) (8) (10) [1] S(T) Ss(T) Sp(T)?

Or am I misreading this?  I'd like the stats on the wiki as close to the coded dice on the site as possible.

**Comment by blackshadowshade** [comment]
I believe that this is not yet agreed, and won't be until I actually try to implement select dice.

**Comment by TheOrgg** [comment]
I had a thought on notation for this issue.  For multiple pools, could we have a notation like "S.1" and "S.2" that limit that selection to the specific dice frame, and then a generic S that could be used for any Select frame?  I'm thinking forward to Fanatics territory here, and the mixed Option dice I've seen on the Eli BM.

So how the behavior would work is that a S( die would be usable in any Select spot, and the S.# ones would only be selectable for the specific spot.  So instead of Eli being (R!/S!) (T!/U!) (V?/X?) (W!/Y!) (Z) he would be written as S.1(R!) S.1(S!) S.2(T!) S.2(U!) S.3(V?) S.3(X?) S.4(W!) S.4(Y!) (Z).   This method would also allow for "Multi-option" dice if someone wanted to do option dice with more than two choices.

Why have a non-limited S(#) die?  For Reserve Select dice to function (a.k.a. Sideboard Dice).

The nomenclature could also be used for #507  (Deception Dice), so ~.4(12) would be a d12 worth four points.

**Comment by jl8e** [comment]
So, that notation is virtually illegible to the player. It's a huge mishmash of swing types, numbers, and the letter S (which is also a swing type)

This is a problem one is likely to get trying to treat a recipe-level ability as a die skill; it's really only feasible for very simple implementations. If select dice are always fightball-style Take 5, then it's easy. If they're not, it's not.

I've already made my thoughts on the matter known, most recently in #2745 

More importantly, notation is not an important factor in implementation. Any reasonably human-readable notation is not all that hard to parse, but the implementation details are going to drive the notation, not the other way around. (For instance, your suggestion would be doable, but doesn't mesh well with the recipe parser, which currently doesn't need to maintain any state about previous dice, but would suddenly have to keep quite a bit about the select skill in particular.)

**Comment by TheOrgg** [comment]
The mishmash is of a button character made by someone at Steve Jackson Games.  It has option dice that wouldn't function on BMO right now as they've got skills within the options.  Having read your Containers essay on the post you linked, their notation would be {!}(R/S) {!}(T/U) {?}(V/X) {!}(W/Y) (Z), or if swing dice were allowed in Option dice, (R/S)! (T/U)! (V/X)? (W/Y)! (Z).

I was trying to find a potential solution to other buttons being designed with a similar approach, or a way that Option dice with more than two choices or choices with skills attached could be implemented via the Select ability.  Multiple select pools makes sense to me.  The only issue is with Reserve Select dice (a.k.a. Sideboard Dice).    

I had the thought of how different Select pools might be implemented, assuming "S" was the notation for the skill, and wanted to document it in case it was useful.

**Comment by TheOrgg** [comment]
Had another thought on how Select notation might be implemented.  Use a number before a list of dice (similar to what was suggested above).  Then if there's a smaller pool or multiple pools, it isn't hard to discern.  So you could have 5(2, 4, 6, 8, 10, 12, 16, 18, 20) where you'd choose five of the dice, or something like 2(4, 6, 8) 3 (10, 12, 16, 20).  The only thing is when the dice have abilities...
